tests/v2.test.js

```diff
@@ -0,0 +1,338 @@
+import expect from 'expect';
+import fs from 'promise-fs';
+import path from 'path';
+import request from 'request-promise-native';
+import uuid from 'uuid/v1';
+
+const { URLSearchParams } = require('url');
+
+const api = request.defaults({
+  baseUrl: 'https://api.fastly.com/service/6lQpb69X5Qt9YVt2OIMiGG/',
+  headers: {
+    Accept: 'application/json',
+    'Fastly-Key': process.env.FASTLY_API_KEY,
+  },
+});
+
+const domain = 'iotest--iiif.elifesciences.org';
+const baseUrl = `https://${domain}`;
+const http = request.defaults({
+  baseUrl,
+  headers: {
+    'Fastly-Debug': '1',
+    'X-Test-Run': uuid(),
+  },
+  resolveWithFullResponse: true,
+});
+
+const source = path.resolve(__dirname, '../src/v2');
+
+beforeAll(async () => {
+  const version = await api.post('version')
+    .then(response => JSON.parse(response).number);
+
+  const config = [
+    api.post(`version/${version}/backend`).form({
+      hostname: 'fastly-iiif-test.s3.amazonaws.com',
+      name: 'bucket',
+      shield: 'dca-dc-us',
+    }),
+    api.post(`version/${version}/condition`).form({
+      name: 'cats-prefix',
+      statement: 'req.http.X-IIIF-Prefix == "kittehs"',
+      type: 'REQUEST',
+    }).then(() => api.post(`version/${version}/header`).form({
+      name: 'cats-prefix',
+      type: 'request',
+      request_condition: 'cats-prefix',
+      action: 'set',
+      dst: 'url',
+      src: '"/cats/" req.http.X-IIIF-Identifier "?" req.url.qs',
+    })),
+    api.post(`version/${version}/domain`).form({
+      name: domain,
+    }),
+    api.post(`version/${version}/header`).form({
+      name: 'vary-test-run',
+      type: 'cache',
+      action: 'append',
+      dst: 'http.Vary',
+      src: '"X-Test-Run"',
```

Comment:
Results in 

```vcl
if (!beresp.http.Vary) {
  set beresp.http.Vary = "X-Test-Run";
} else {
  set beresp.http.Vary = beresp.http.Vary "X-Test-Run";
}
```

So should include a comma here? (`set beresp.http.Vary = ", X-Test-Run";` when it's not set would be odd, but probably doesn't cause a problem?)

Reply:
Might be able to do something with `regex` otherwise? (Trying to avoid writing VCL directly here.)